prototypes/ItemRepeaterPrototypes/ItemsRepeaterExperiments/AttachedBehaviors/SelectionModelPage.xaml.cs

```diff
@@ -0,0 +1,101 @@
+﻿using ItemsRepeaterExperiments.Common;
+using Microsoft.UI.Xaml.Controls;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+using Windows.UI.Xaml.Data;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Navigation;
+
+// The Blank Page item template is documented at https://go.microsoft.com/fwlink/?LinkId=234238
+
+namespace ItemsRepeaterExperiments.AttachedBehaviors
+{
+    /// <summary>
+    /// An empty page that can be used on its own or navigated to within a Frame.
+    /// </summary>
+    public sealed partial class SelectionModelPage : Page
+    {
+        private List<Team> teams = DataSourceCreator<Team>.CreateRandomizedList(100);
+        public SelectionModelPage()
+        {
+            this.InitializeComponent();
+        }
+
+
+        /// <summary>
+        /// Attach properties to virtualized items and update containers
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void ItemsRepeater_ElementPrepared(ItemsRepeater sender, ItemsRepeaterElementPreparedEventArgs args)
+        {
+            var control = args.Element as FrameworkElement;
+
+            // Attach repeater and selectionmodel properties for handling of selection
+            SelectionBehavior.SetParentItemsRepeater(control, TeamPresenter);
+            SelectionBehavior.SetSelectionModel(control, selectionModel);
+
+            SelectionBehavior.SetIsSelected(control,
+                selectionModel.SelectedItems.Contains(teams[TeamPresenter.GetElementIndex(control)]));
+        }
+
+        /// <summary>
+        /// This event gets raised when an item gets selected
+        /// @WINUI This event definitely needs a list of selected and deselected items
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="args"></param>
+        private void SelectionModel_SelectionChanged(SelectionModel sender, SelectionModelSelectionChangedEventArgs args)
```

Comment:
>SelectionModel_SelectionChanged [](start = 21, length = 31)

This could be on SelectionBehavior as well and be specific to each element we attached it to. If we did that we would automatically do it for only for the realized items.

Reply:
For now we could do exactly what you are doing here in the behavior. If we had access to the list of realized elements, we could do it just for those. Once we have more information in selection changed event, we can leverage those to update only the changed elements.

Reply:
The problem I see is where do we get the ItemsRepeater from? Unfortunately (or luckily) SelectionModel is not a DP so we can't set a property for that. And having a dictionary inside the behavior to store does not seem that great of an idea.